mantidproject/mantid
# New algorithm to slice event workspaces for ISIS reflectometry

**Description of work.**
This PR adds a new algorithm to perform slicing of event workspaces, producing output suitable for use with ReflectometryReductionOneAuto.
- Uses GenerateEventsFilter and FilterEvents to do the filtering
- Rebins to histogram data
- Adds monitors to each resulting slice, scaled by the relative proton_charge for the slice

The aim of this algorithm will eventually be to replace functionality currently implemented in the ISIS Reflectometry GUI, which should simplify the restructuring work in #23051.

**Report to:** nobody

**To test:**
Try out the algorithm with various different inputs for filtering, e.g. the following should result in a group workspace with 3 slices. Check that each contains 3 monitors with scaled Y values compared to the input monitor workspace.
```
Load('OFFSPEC48276', OutputWorkspace='OFFSPEC48276',LoadMonitors=True)
ReflectometrySliceEventWorkspace(InputWorkspace='OFFSPEC48276', MonitorWorkspace='OFFSPEC48276_monitors',OutputWorkspace='TOF_48276',
    TimeInterval=800)
```

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
